woocommerce/woocommerce
# SelectTree: allow navigation between items and input using tab and arrow keys

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This fixes some things that were outright missing from the SelectTree control regarding keyboard usability. As far as I know, there is no issue created for this, but I took the liberty of implementing this, as it seems pretty obvious.

I also added a label to the clear button (Remove all), as it wasn't accessible, since it was only an icon.

I also fixed a small issue in the "Create new category" dialog, as the help text showing up should only be shown for multiple mode.




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
2. Go to Products > Add
3. Go to the Organization tab
4. Select some categories
5. While focusing on the input, press the left arrow key and check that it's possible to navigate between them. 
6. While on the items, press the right arrow key until you reach the end of the list and verify that the focus switches to the input



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
